src/Eloquent/Model.php

```diff
@@ -57,4 +58,62 @@ final public static function isDocumentModel(string|object $class): bool
         // Document models must use the DocumentModel trait.
         return self::$documentModelClasses[$class] = array_key_exists(DocumentModel::class, class_uses_recursive($class));
     }
+
+    /**
+     * Override of Laravel's performInsert() to handle MongoDB _id immutability.
+     *
+     * Ensures syncOriginal() is called immediately after insert, so that
+     * subsequent save() calls (e.g., inside the "created" model event)
+     * don't treat the primary key (_id) as dirty.
+     *
+     * @param Builder $query
+     *
+     * @return bool
+     */
+    protected function performInsert(Builder $query)
```

Comment:
Overriding and duplicating a big method like this makes the code more difficult to maintain with Laravel updates. I would prefer to avoid.